### PR TITLE
fix #224 Support Chinese

### LINKED ISSUE #131: Url file name Chinese error
No file was found after the URL was encoded

image
image
#196

### LINKED ISSUE #224: [Bug]Doesn't work when path contains Chinese characters
> File not found
> The file "d:\Code\WebFrontEnd\CSSLearning\03Ã¥Â…ÂƒÃ§Â´Â Ã¦Â˜Â¾Ã§Â¤ÂºÃ¦Â¨Â¡Ã¥Â¼Â.html" cannot be found. It may have been moved, edited, or deleted.

### PR DESCRIPTION
**Pull request by abliger**
1. Open the Chinese path file normally
2. The address bar decode
3. 404 path

Fixes #131

### CODE REVIEW

File: src/server/httpServer.ts [hunk 1]
```diff
@@ -158,7 +158,10 @@ export class HttpServer extends Disposable {
 
 		if (!basePath && (URLPathName == '/' || URLPathName == '')) {
 			const respInfo = this._contentLoader.createNoRootServer();
-			res.writeHead(404);
+			res.writeHead(404, {
+				'Accept-Ranges': 'bytes',
+				'Content-Type': `${respInfo.ContentType}; charset=UTF-8`,
```
Comment (andreamah) [inline src/server/httpServer.ts:163]: I'm not too sure why this is needed on the 404 response?

File: src/server/httpServer.ts [hunk 2]
```diff
@@ -167,7 +170,8 @@ export class HttpServer extends Disposable {
 		}
 
 		let looseFile = false;
-		let absoluteReadPath = path.join(basePath ?? '', unescape(URLPathName));
+		let absoluteReadPath = path.join(basePath ?? '', decodeURI(URLPathName));
+		URLPathName = decodeURI(URLPathName)
```
Comment (andreamah) [inline src/server/httpServer.ts:174]: nit: you can put this line above line 173 and then you don't need to use `decodeURI` within the `path.join` args.

Comment (andreamah) [inline src/server/httpServer.ts:174]:
```suggestion
		URLPathName = decodeURI(URLPathName);

```

**Review by andreamah — CHANGES_REQUESTED** [review]
Thanks so much for the contribution and apologies for getting to it so late!

File: src/server/httpServer.ts (hunk 1 above)
Comment (abliger) [reply src/server/httpServer.ts:163]: image
for relativePath

File: src/server/httpServer.ts (hunk 2 above)
Comment (abliger) [reply src/server/httpServer.ts]: oh!! I didn't notice

File: src/server/httpServer.ts (hunk 1 above)
Comment (andreamah) [reply src/server/httpServer.ts:163]: Ahh I see, it's for when you enter a special character and are given a 404.

**Review by andreamah — APPROVED** [review]

**Review by connor4312 — APPROVED** [review]

**Pull request merged**